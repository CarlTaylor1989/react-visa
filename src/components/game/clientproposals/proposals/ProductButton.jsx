import React from 'react';
import { I18n, Translate } from 'react-redux-i18n';
import PropTypes from 'prop-types';

import './ProductButton.scss';

const ProductButton = ({
  client,
  disabled,
  isSelected,
  id,
  productId,
  onProductChange,
  onLearnMore
}) => {
  const familyId = I18n.t(`clientprops.clients.${client}.products.${id}.familyId`);
  const selectedClass = isSelected ? ' selected' : '';
  const extraClasses = disabled ? ` disabled${selectedClass}` : selectedClass;

  return (
    <div className={`productButton${extraClasses}`}>
      <button
        type="button"
        className={`productButtonWrap ${familyId}`}
        disabled={disabled}
        onClick={() => onProductChange(productId)}
      >
        <div className="productInfo">
          <div className="productTitles">
            <Translate
              className="productFamily"
              value={`clientprops.clients.${client}.products.${id}.familyName`}
            />
            <Translate
              className="productName"
              value={`clientprops.clients.${client}.products.${id}.productName`}
            />
          </div>
        </div>
      </button>
      {onLearnMore && onLearnMore !== null && (
        <button
          className="productLearnCta"
          disabled={disabled}
          onClick={() => onLearnMore(id)}
          type="button"
        >
          <Translate
            className="text"
            value="clientprops.solutionsSet.solutionsLearnBtn"
          />
          <span className="arrowRight" />
        </button>
      )}
    </div>
  );
};

ProductButton.defaultProps = {
  disabled: false,
  onLearnMore: null,
  onProductChange: null
};

ProductButton.propTypes = {
  client: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  isSelected: PropTypes.bool.isRequired,
  id: PropTypes.string.isRequired,
  productId: PropTypes.string.isRequired,
  onProductChange: PropTypes.func,
  onLearnMore: PropTypes.func
};

export default ProductButton;
