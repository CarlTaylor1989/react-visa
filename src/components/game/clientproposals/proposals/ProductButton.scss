@import '../../../../styles/buttons';
@import '../../../../styles/generic';

$arrow-right: '../../../../styles/icons/clients/arrow_right.svg';
$arrow-right-width: 8px;
$arrow-right-height: 14px;

.productButton {
  display: inline-block;
  margin-bottom: 15px;
  margin-right: 15px;

  &:last-child,
  &:nth-last-child(2) {
    margin-bottom: 0;

    &:nth-child(odd) {
      margin-bottom: 0;
    }
  }

  &:nth-child(even) {
    margin-right: 0;
  }

  .productButtonWrap {
    @extend %button-reset;
    @include align-items(center);
    @include border-radius(16px);
    background-color: $colour-blue;
    border: 2px solid $colour-white;
    box-sizing: border-box;
    color: $colour-white;
    cursor: pointer;
    font-size: 16px;
    position: relative;
    transition: background-color .25s ease-in-out;
    width: 170px;

    &:hover {
      &.family1 {
        background-color: $colour-family-1-sol-btn-hover;
      }

      &.family2 {
        background-color: $colour-family-2-sol-btn-hover;
      }

      &.family3 {
        background-color: $colour-family-3-sol-btn-hover;
      }

      &.family4 {
        background-color: $colour-family-4-sol-btn-hover;
      }

      &.family5 {
        background-color: $colour-family-5-sol-btn-hover;
      }
    }

    .productInfo {
      height: 150px;
      padding: 10px 15px;

      .productFamily,
      .productName {
        display: block;
        font-weight: 700;
        text-align: left;
      }

      .productFamily {
        font-size: 14px;
      }

      .productName {
        font-size: 16px;
        padding-bottom: 5px;
      }
    }
  }

  .productLearnCta {
    @extend %button-reset;
    @include align-items(center);
    color: $colour-white;
    cursor: pointer;
    display: flex;
    padding: 10px 0 0 40px;

    .text {
      font-size: 14px;
      font-weight: 600;
    }

    .arrowRight {
      background-image: url($arrow-right);
      display: block;
      height: $arrow-right-height;
      margin-left: 10px;
      width: $arrow-right-width;
    }

    &:hover {
      text-decoration: underline;
    }
  }

  &.selected {
    .productButtonWrap {
      &.family1 {
        @include box-shadow(0, 0, 14px, 0, $colour-family-1-sol-btn);
        background-color: $colour-family-1-sol-btn;
      }

      &.family2 {
        @include box-shadow(0, 0, 14px, 0, $colour-family-2-sol-btn);
        background-color: $colour-family-2-sol-btn;
      }

      &.family3 {
        @include box-shadow(0, 0, 14px, 0, $colour-family-3-sol-btn);
        background-color: $colour-family-3-sol-btn;
      }

      &.family4 {
        @include box-shadow(0, 0, 14px, 0, $colour-family-4-sol-btn);
        background-color: $colour-family-4-sol-btn;
      }

      &.family5 {
        @include box-shadow(0, 0, 14px, 0, $colour-family-5-sol-btn);
        background-color: $colour-family-5-sol-btn;
      }
    }
  }

  &.disabled {
    .productButtonWrap,
    .productLearnCta {
      cursor: not-allowed;

      &:hover {
        text-decoration: none;
      }
    }

    &:not(.selected) {
      .productButtonWrap {
        &:hover {
          background-color: $colour-blue;
        }
      }
    }
  }
}
