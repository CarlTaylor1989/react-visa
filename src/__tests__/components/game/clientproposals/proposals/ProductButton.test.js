/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import ProductButton from
  '../../../../../components/game/clientproposals/proposals/ProductButton';

let mockProps;
beforeEach(() => {
  mockProps = {
    client: 'client1',
    id: '0',
    isSelected: false,
    productId: 'prod1',
    onProductChange: jest.fn(),
    onLearnMore: jest.fn()
  };
});

test('renders the ProductButton component with the selected state', () => {
  const wrapper = shallow(<ProductButton {...mockProps} isSelected />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('renders the ProductButton component with the unselected state', () => {
  const wrapper = shallow(<ProductButton {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('renders the ProductButton component with the selected disabled state', () => {
  const wrapper = shallow(
    <ProductButton
      {...mockProps}
      isSelected
      disabled
    />
  );
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('renders the ProductButton component with the unselected disabled state', () => {
  const wrapper = shallow(<ProductButton {...mockProps} disabled />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('button calls onProductChange() method when selected', () => {
  const wrapper = shallow(<ProductButton {...mockProps} />);
  wrapper.find('.productButtonWrap').at(0).simulate('click');
  expect(mockProps.onProductChange).toHaveBeenCalled();
});

test('learn more calls onLearnMore() method when selected', () => {
  const wrapper = shallow(<ProductButton {...mockProps} />);
  wrapper.find('.productLearnCta').at(0).simulate('click');
  expect(mockProps.onLearnMore).toHaveBeenCalled();
});
