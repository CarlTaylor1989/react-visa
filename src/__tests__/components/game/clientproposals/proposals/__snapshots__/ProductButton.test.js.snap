// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ProductButton component with the selected disabled state 1`] = `
<div
  className="productButton disabled selected"
>
  <button
    className="productButtonWrap familyId"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <div
      className="productInfo"
    >
      <div
        className="productTitles"
      >
        <Translate
          className="productFamily"
          tag="span"
          value="clientprops.clients.client1.products.0.familyName"
        />
        <Translate
          className="productName"
          tag="span"
          value="clientprops.clients.client1.products.0.productName"
        />
      </div>
    </div>
  </button>
  <button
    className="productLearnCta"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <Translate
      className="text"
      tag="span"
      value="clientprops.solutionsSet.solutionsLearnBtn"
    />
    <span
      className="arrowRight"
    />
  </button>
</div>
`;

exports[`renders the ProductButton component with the selected state 1`] = `
<div
  className="productButton selected"
>
  <button
    className="productButtonWrap familyId"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <div
      className="productInfo"
    >
      <div
        className="productTitles"
      >
        <Translate
          className="productFamily"
          tag="span"
          value="clientprops.clients.client1.products.0.familyName"
        />
        <Translate
          className="productName"
          tag="span"
          value="clientprops.clients.client1.products.0.productName"
        />
      </div>
    </div>
  </button>
  <button
    className="productLearnCta"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <Translate
      className="text"
      tag="span"
      value="clientprops.solutionsSet.solutionsLearnBtn"
    />
    <span
      className="arrowRight"
    />
  </button>
</div>
`;

exports[`renders the ProductButton component with the unselected disabled state 1`] = `
<div
  className="productButton disabled"
>
  <button
    className="productButtonWrap familyId"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <div
      className="productInfo"
    >
      <div
        className="productTitles"
      >
        <Translate
          className="productFamily"
          tag="span"
          value="clientprops.clients.client1.products.0.familyName"
        />
        <Translate
          className="productName"
          tag="span"
          value="clientprops.clients.client1.products.0.productName"
        />
      </div>
    </div>
  </button>
  <button
    className="productLearnCta"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <Translate
      className="text"
      tag="span"
      value="clientprops.solutionsSet.solutionsLearnBtn"
    />
    <span
      className="arrowRight"
    />
  </button>
</div>
`;

exports[`renders the ProductButton component with the unselected state 1`] = `
<div
  className="productButton"
>
  <button
    className="productButtonWrap familyId"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <div
      className="productInfo"
    >
      <div
        className="productTitles"
      >
        <Translate
          className="productFamily"
          tag="span"
          value="clientprops.clients.client1.products.0.familyName"
        />
        <Translate
          className="productName"
          tag="span"
          value="clientprops.clients.client1.products.0.productName"
        />
      </div>
    </div>
  </button>
  <button
    className="productLearnCta"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <Translate
      className="text"
      tag="span"
      value="clientprops.solutionsSet.solutionsLearnBtn"
    />
    <span
      className="arrowRight"
    />
  </button>
</div>
`;
